lib/facter/powershell_version.rb

```diff
@@ -0,0 +1,49 @@
+Facter.add(:powershell_installed) do
+ confine :operatingsystem => :windows
+ require 'win32/registry'
+ setcode do
+  begin
+    Win32::Registry::HKEY_LOCAL_MACHINE.open('SOFTWARE\Microsoft\PowerShell\1\PowerShellEngine', ::Win32::Registry::KEY_READ)
+    true
+  rescue
+    false
+  end
+ end
+end
+
+Facter.add(:powershell_version) do
+  confine :operatingsystem => :windows, :powershell_installed => true
+  require 'win32/registry'
+  setcode do
+    Win32::Registry::HKEY_LOCAL_MACHINE.open('SOFTWARE\Microsoft\PowerShell\3\PowerShellEngine') do |reg|
```

Comment:
This key will be in both x64 and Wow6432Node, so I don't know if we need to turn off registry redirection (the flags `Win32::Registry::KEY_READ | 0x100`). Prolly for safety we should.